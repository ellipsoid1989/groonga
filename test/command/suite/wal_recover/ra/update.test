# _database_close/_database_reopen can't use with HTTP because requests via HTTP are always
# processed by workers. Workers never use primary WAL role.
#@require-interface stdio

#@require-feature message_pack

#$GROONGA_ALLOW_DATABASE_REOPEN=yes
#$GRN_WAL_ROLE=primary

#@disable-logging
cache_limit --max 0
#@enable-logging

table_create memos TABLE_NO_KEY
column_create memos count COLUMN_SCALAR UInt32

load --table memos
[
{"count": 10},
{"count": 20},
{"count": 30}
]

io_flush
#@copy-path #{db_path}.0000101 #{db_path}.0000101.loaded

load --table memos
[
{"_id": 2, "count": 200}
]

#@copy-path #{db_path}.0000101.wal #{db_path}.0000101.wal.keep
_database_close

#@copy-path #{db_path}.0000101.loaded #{db_path}.0000101
#@copy-path #{db_path}.0000101.wal.keep #{db_path}.0000101.wal
_database_reopen

select memos
