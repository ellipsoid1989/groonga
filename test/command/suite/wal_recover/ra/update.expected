table_create memos TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create memos count COLUMN_SCALAR UInt32
[[0,0.0,0.0],true]
load --table memos
[
{"count": 10},
{"count": 20},
{"count": 30}
]
[[0,0.0,0.0],3]
io_flush
[[0,0.0,0.0],true]
load --table memos
[
{"_id": 2, "count": 200}
]
[[0,0.0,0.0],1]
_database_close
[[0,0.0,0.0],true]
_database_reopen
[[0,0.0,0.0],true]
select memos
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "count",
          "UInt32"
        ]
      ],
      [
        1,
        10
      ],
      [
        2,
        200
      ],
      [
        3,
        30
      ]
    ]
  ]
]
